source/code/plugin/health/node_monitor_hierarchy_reducer.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module HealthModel
+  class NodeMonitorHierarchyReducer
+    def initialize
+    end
+
+    # Finalizes the Node Hierarchy. This removes node pools and node pool set from the hierarchy if they are not present.
+    def finalize(monitor_set)
+      monitors_to_reduce = [MonitorId::ALL_AGENT_NODE_POOLS, MonitorId::ALL_NODES]
+      # for the above monitors, which are constant per cluster, the monitor_id and monitor_instance_id are the same
+      monitors_to_reduce.each do |monitor_to_reduce|
+        monitor = monitor_set.get_monitor(monitor_to_reduce)
+        if !monitor.nil?
+            puts "Before Deleting #{monitor_set.get_size}"
```

Comment:
this will go to stdout and we will pick this up. please remove.

Reply:
Done